derive_builder_core/src/macro_options/darling_opts.rs

```diff
@@ -95,6 +95,25 @@ pub struct BuildFn {
     /// * If `validate` is specified, then this type must provide a conversion from the specified
     ///   function's error type.
     error: Option<Path>,
+    /// Whether to generate `ValidationError(String)` as a variant of the build error type.
+    ///
+    /// Setting this to `false` will prevent `derive_builder` from using the `validate` function
+    /// but this also means it does not generate any usage of the `alloc` crate (useful when
+    /// disabling the `alloc` feature in `no_std`).
+    validation_error: bool,
+}
+
+impl BuildFn {
+    fn validation_needs_error(self) -> darling::Result<Self> {
+        if self.validate.is_some() && !self.validation_error {
+            Err(darling::Error::custom(
+                "Cannot set `validation_error = false` when using `validate`",
+            )
+            .with_span(&self.validate))
```

Comment:
The wording of the error message feels incorrectly paired with its span; I would either make the message be "Cannot use `validate` when `validate_error = false`" or I would set the span to be the `validate_error` field.

Reply:
Agreed, I changed the span to point to `validation_error` now.